cmd/frontend/graphqlbackend/search_alert.go

```diff
@@ -750,6 +753,10 @@ func (o *alertObserver) Done(stats *streaming.Stats) (*searchAlert, error) {
 		o.update(alertForStructuralSearchNotSet(o.Inputs.OriginalQuery))
 	}
 
+	if !o.hasResults && stats.Status.All(search.RepoStatusTimedout) && stats.Status.Len() == len(stats.Repos) {
+		return alertForTimeout(o.Inputs), nil
```

Comment:
Will "timed out" always be the highest priority alert? This is currently overriding any other alerts. Maybe just update instead, and make `alertForTimeout` high priority?

```suggestion
		o.update(alertForTimeout(o.Inputs))
```